src/System.CommandLine.Tests/CompletionTests.cs

```diff
@@ -41,9 +41,9 @@ public void Command_GetCompletions_returns_available_option_aliases()
         {
             var command = new Command("command")
             {
-                new Option<string>("--one", "option one"),
-                new Option<string>("--two", "option two"),
-                new Option<string>("--three", "option three")
+                new Option<string>("--one") { Description = "option one" },
+                new Option<string>("--two") { Description = "option two" },
+                new Option<string>("--three") { Description = "option three" },
```

Comment:
The change to the `params` constructor for aliases results in some extra verbosity here and I wonder if it's the right tradeoff, given that people using `Option` set a description much more often than aliases.

Reply:
We have at least three ways of setting the aliases:
* what we had so far, which required the user to allocate an array in an explicit way
```cs
new Option<bool>("--help", new [] { "-h", "-?" }, "description");
```
* using params, which looks way nicer, but requires the description to be set via a property. TBH I am not sure if it's a bad thing, but it's very subjective.
```cs
new Option<bool>("--help", "-h", "-?")
{
    Description = "description"
};
```
* using collection initalizer syntax. IMO it looks great but I would expect very few C# devs to know about this duck typing.
```cs
new Option<bool>("--help", "description")
{
    Aliases = { "-h", "-?" }
};
```

@jonsequitur if you believe it's worth it, I can remove the params ctor and revert this part of the changes. My personal preference would be to merge it in the new form and eventually revisit it after API design review.

Reply:
My hesitation is that the more common case is now more verbose, and it will be a very large source-breaking change. The reasoning makes sense though. I'd be curious to hear more opinions.

Reply:
> I'd be curious to hear more opinions.

@KathleenDollard @Keboo @KalleOlaviNiemitalo what is your opinion on that?

Reply:
I am going to merge the PR now, but please don't hesitate to share your feedback.

Reply:
In my first example above, I meant to explicitly state that there'd still be the single-string overload too. Calling that out in case it wasn't inferred.

```c#
Option(string name, string description);
Option(string[] nameAndAliases, string description);
```